### PR TITLE
ARM spec adjust

### PR DESCRIPTION
**Pull request by mjudeikis**
### Which issue this PR addresses:

~~https://github.com/Azure/ARO-RP/pull/1654 MUST GO FIRST~~

https://github.com/Azure/azure-rest-api-specs/pull/15616 is source of all these changes

This adjust some of the field handling we do for ARM spec to match 

### What this PR does / why we need it:

ARM validation fails so we need to adjust this a bit.

### Test plan for issue:

No

### Is there any documentation that needs to be updated for this PR?

No

### CODE REVIEW

File: pkg/swagger/typewalker.go [hunk 1]
```diff
@@ -88,6 +88,7 @@ func (tw *typeWalker) schemaFromType(t types.Type, deps map[*types.Named]struct{
 			s.Type = "boolean"
 		case types.Int:
 			s.Type = "integer"
+			s.Format = "int64"
```
Comment (m1kola) [inline pkg/swagger/typewalker.go:91]: Why this change? It does seem to affect specs for older versions. Is it expected?

File: pkg/swagger/typewalker.go [hunk 2]
```diff
@@ -133,17 +136,29 @@ func (tw *typeWalker) schemaFromType(t types.Type, deps map[*types.Named]struct{
...
 		}
 
 	default:
 		panic(t)
 	}
 
+	//for _, xname := range tw.xmsSecretList {
+
+	//}
+
```
Comment (troy0820) [inline pkg/swagger/typewalker.go:161]:
```suggestion

```
Can this be deleted?

File: swagger/redhatopenshift/resource-manager/Microsoft.RedHatOpenShift/preview/2021-09-01-preview/examples/OpenShiftClusters_CreateOrUpdate.json [hunk 3]
```diff
@@ -6,14 +6,6 @@
...
     "parameters": {
       "location": "location",
-      "systemData": {
-        "createdBy": "string",
-        "createdByType": "Application",
-        "createdAt": "2020-02-03T01:01:01.1075056Z",
-        "lastModifiedBy": "string",
-        "lastModifiedByType": "Application",
-        "lastModifiedAt": "2020-02-03T01:01:01.1075056Z"
-      },
```
Comment (troy0820) [inline swagger/redhatopenshift/resource-manager/Microsoft.RedHatOpenShift/preview/2021-09-01-preview/examples/OpenShiftClusters_CreateOrUpdate.json:16]: Did this get deleted during `make client`?  Is it supposed to be deleted?  I know it's changed to a pointer but I didn't believe it would be deleted.

File: swagger/redhatopenshift/resource-manager/Microsoft.RedHatOpenShift/preview/2021-09-01-preview/examples/OpenShiftClusters_Update.json [hunk 4]
```diff
@@ -5,14 +5,6 @@
...
     "resourceName": "resourceName",
     "parameters": {
-      "systemData": {
-        "createdBy": "string",
-        "createdByType": "Application",
-        "createdAt": "2020-02-03T01:01:01.1075056Z",
-        "lastModifiedBy": "string",
-        "lastModifiedByType": "Application",
-        "lastModifiedAt": "2020-02-03T01:01:01.1075056Z"
-      },
```
Comment (troy0820) [inline swagger/redhatopenshift/resource-manager/Microsoft.RedHatOpenShift/preview/2021-09-01-preview/examples/OpenShiftClusters_Update.json:15]: Same as above.

File: pkg/swagger/typewalker.go (hunk 1 above)
Comment (mjudeikis) [reply pkg/swagger/typewalker.go]: Removed. See original PR documented in the description

File: swagger/redhatopenshift/resource-manager/Microsoft.RedHatOpenShift/preview/2021-09-01-preview/examples/OpenShiftClusters_CreateOrUpdate.json (hunk 3 above)
Comment (mjudeikis) [reply swagger/redhatopenshift/resource-manager/Microsoft.RedHatOpenShift/preview/2021-09-01-preview/examples/OpenShiftClusters_CreateOrUpdate.json:16]: See original PR link in the description

**Review by m1kola — APPROVED** [review]
